jmsSource/REAMDE.md

```diff
@@ -0,0 +1,321 @@
+myObj.JMSFormat = msg.JMSFormat
+myObj.JMSDestination = msg.destination
+INSERT JMSMessageType(myObj)
+```
+
+## Select Statements
+
+In order to read messages from a queue, (**NOT** a queueListener), a VAIL Select statement must be used. The Select statement 
+must have two query parameters: *operation* and *queue*. Currently there is only one *operation* that is supported, which is 
+the "read" operation. In the future, there may be different Select Operations. The *queue* parameter is the name of the queue
+from which to read. The following is an example of a Procedure created in VANTIQ Modelo querying against a JMS Source:
+
+```
+PROCEDURE readMessageFromQueue()
+
+SELECT * FROM SOURCE JMS1 AS msg WITH
+	operation: "read",
+	queue: "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dq"
+  
+  {
+      var myObj = {}
+      myObj.JMSMessage = msg.message
+      myObj.JMSFormat = msg.JMSFormat
+      myObj.JMSDestination = msg.destination
+      INSERT JMSMessageType(myObj)
+  }
+```
+
+*   **NOTE**: Messages returned by a Select statement are in the exact same format as those sent to the source as a Notification.
+
+## Publish Statements
+
+In order to send messages to either a queue or a topic, a VAIL Publish statement must be used. The Publish statement will have
+three parameters: *message*, *queue/topic*, *JMSFormat*. The *message* parameter is simply the message to be sent. The next
+parameter is either *queue* or *topic*, depending on which type of destination you are sending the message to. The value of this
+parameter is the name of the given queue/topic. Finally, the last parameter is the *JMSFormat*, which specifies the JMS Message
+Type that will be used to send the message. If no custom Message Handler was implemented, the three values that we currently
+support are "TextMessage", "MapMessage", and "Message". The "Message" type does not have any message body, so no message parameter
+is needed. The following two examples show how to send all three supported message types.
+
+### Sending Text Message ###
+
+```
+PROCEDURE sendTextMessage()
+
+var msg = "A message sent from VANTIQ to a topic."
+var dest = "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dt"
+var msgFormat = "TextMessage"
+
+PUBLISH {message: msg, topic: dest, JMSFormat: msgFormat} to SOURCE JMS1 
+```
+
+### Sending Map Message ###
+```
+PROCEDURE sendMapMessage()
+
+var msg = {}
+msg.age = 22
+msg.name = "Namir"
+var dest = "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dq"
+var msgFormat = "MapMessage"
+
+PUBLISH {message: msg, queue: dest, JMSFormat: msgFormat} to SOURCE JMS1 
+```
+
+### Sending Message ###
+```
+PROCEDURE sendMapMessage()
+
+var dest = "NamirJMSServer-0/NamirSystemModule-0!NamirJMSServer-0@/com/namir/weblogic/base/dq"
+var msgFormat = "Message"
+
+PUBLISH {queue: dest, JMSFormat: msgFormat} to SOURCE JMS1 
+```
+
+## Custom Message Handlers ##
+
```

Comment:
Reiterate purpose of message handlers

We may also want to note that this section is for those for developers of message handlers, not configurers of sources.  We might want to have some overview section at the top to identify these roles?